presto-main/src/test/java/com/facebook/presto/operator/scalar/sql/TestIPSqlFunctions.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar.sql;
+
+import com.facebook.presto.operator.scalar.AbstractTestFunctions;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+
+public class TestIPSqlFunctions
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testIsPrivateTrue()
+    {
+        /*
+         * IPv4
+         */
+
+        // 0.0.0.0/8 RFC1122: "This host on this network"
+        assertFunction(
```

Comment:
Can we refactor this to use a `org. testng. annotation.DataProvider` for the tests cases instead ?

Reply:
Ah, the generated test per input is really nice.